pkg/logs/input/docker/launcher.go

```diff
@@ -151,12 +97,12 @@ func (l *Launcher) run() {
 		select {
 		case service := <-l.addedServices:
 			// detected a new container running on the host,
-			cli, err := l.getCli()
+			dockerutil, err := dockerutil.GetDockerUtil()
```

Comment:
do we want to keep a reference to dockerutil instead of calling it for each new service ? What can go wrong ?

Reply:
I’ve introduced the `l.getCli()` earlier in order to be more resilient to situations where the agent is started before it is able to connect to the docker daemon.
In those situations, we might get a non-`nil` error and a `nil` client.
So, we would need to cache the client when it succeeds to connect to the docker daemon and to retry later when it fails.
This was the purpose of the `retrier` that there was inside the `getCli()`.
That retrier is now inside `GetDockerUtil()`: https://github.com/DataDog/datadog-agent/blob/4cc1fe3ca3f14aa752b63f0c044c01a9029a86bc/pkg/util/docker/global.go#L40-L44

The caching logic is in fact inside `GetDockerUtil()`.

Reply:
sounds good, thanks for the explanation